label_studio/fsm/models.py

```diff
@@ -194,3 +196,148 @@ def _get_entity_field_name(cls) -> str:
+            # Index with DESC order explicitly supports ORDER BY id DESC queries
+            models.Index(fields=['task_id', '-id'], name='task_current_state_idx'),
+            # Reporting and filtering
+            models.Index(fields=['project_id', 'state', '-id'], name='task_project_state_idx'),
+            models.Index(fields=['organization_id', 'state', '-id'], name='task_org_reporting_idx'),
+            # History queries
+            models.Index(fields=['task_id', 'id'], name='task_history_idx'),
+        ]
+        # No constraints needed - INSERT-only approach
+        ordering = ['-id']
+
+    @classmethod
+    def get_denormalized_fields(cls, entity):
+        """Get denormalized fields for TaskState creation"""
+        return {
+            'project_id': entity.project_id,
+        }
+
+    @property
+    def is_terminal_state(self) -> bool:
+        """Check if this is a terminal task state"""
+        return self.state == TaskStateChoices.COMPLETED
+
+
+@register_state_model('annotation')
+class AnnotationState(BaseState):
+    """
+    Core annotation state tracking for Label Studio.
+    Provides basic annotation state management with:
+    - Simple 3-state workflow (DRAFT → SUBMITTED → COMPLETED)
+    """
+
+    # Entity Relationship
+    annotation = models.ForeignKey('tasks.Annotation', on_delete=models.CASCADE, related_name='fsm_states')
+
+    # Override state field to add choices constraint
+    state = models.CharField(max_length=50, choices=AnnotationStateChoices.choices, db_index=True)
+
+    # Denormalized fields for performance (avoid JOINs in common queries)
+    task_id = models.PositiveIntegerField(
+        db_index=True, help_text='From annotation.task_id - denormalized for performance'
+    )
+    project_id = models.PositiveIntegerField(
+        db_index=True, help_text='From annotation.task.project_id - denormalized for performance'
+    )
+    completed_by_id = models.PositiveIntegerField(
+        null=True, db_index=True, help_text='From annotation.completed_by_id - denormalized for performance'
+    )
+
+    class Meta:
+        app_label = 'fsm'
+        indexes = [
+            # Critical: Latest state lookup
+            models.Index(fields=['annotation_id', '-id'], name='anno_current_state_idx'),
+            # Filtering and reporting
+            models.Index(fields=['task_id', 'state', '-id'], name='anno_task_state_idx'),
+            models.Index(fields=['completed_by_id', 'state', '-id'], name='anno_user_report_idx'),
+            models.Index(fields=['project_id', 'state', '-id'], name='anno_project_report_idx'),
+        ]
+        ordering = ['-id']
+
+    @classmethod
+    def get_denormalized_fields(cls, entity):
+        """Get denormalized fields for AnnotationState creation"""
+        return {
+            'task_id': entity.task.id,
+            'project_id': entity.task.project_id,
+            'completed_by_id': entity.completed_by.id if entity.completed_by else None,
```

Comment:
```suggestion
            'completed_by_id': entity.completed_by_id if entity.completed_by_id else None,
```
This saves a query to get the user